sklearn/linear_model/tests/test_coordinate_descent.py

```diff
@@ -1280,6 +1280,15 @@ def test_enet_l1_ratio():
     with pytest.raises(ValueError, match=msg):
         MultiTaskElasticNetCV(l1_ratio=0, random_state=42).fit(X, y[:, None])
 
+    # Test that l1_ratio=0 with alpha>0 produces user warning
+    warning_message = (
+        "Coordinate descent without L1 regularization may "
+        "lead to unexpected results and is discouraged. "
+        "Set l1_ratio > 0 to add L1 regularization."
+    )
+    with pytest.warns(UserWarning, match=warning_message):
+        ElasticNetCV(l1_ratio=[0], alphas=[1]).fit(X, y)
```

Comment:
Actually, this test can be improved as to check that the warning is raised at fit time.
```suggestion
    )
    est = ElasticNetCV(l1_ratio=[0], alphas=[1])
    with pytest.warns(UserWarning, match=warning_message):
        est.fit(X, y)
```